### ISSUE TITLE
Markdown syntax highlighting fails after ```

### ISSUE THREAD

**Issue by peterbe** [atom/language-gfm#21]
![screenshot 2014-04-21 11 01 17](https://cloud.githubusercontent.com/assets/26739/2780153/ccaf384c-cb03-11e3-958a-497fe2ab5abd.png)

It's reproducible every time with [this file](https://raw.githubusercontent.com/peterbe/grymt/master/README.md) in Atom 0.90.0

**Comment by plttn** [comment]
It still seems to be breaking. even just a basic `<html></html>` pairing causes the rest of the file to be treated with the HTML grammar. repro file: https://raw.githubusercontent.com/plttn/boilerplates/master/markdown.md

**Comment by plttn** [comment]
It still seems to be breaking. https://raw.githubusercontent.com/plttn/boilerplates/master/markdown.md this file will syntax highlight properly, but snippets below the fenced in block will be using the HTML grammar vs the GFM grammar.

**Comment by ArtskydJ** [comment]
Got the same issue in atom v1.0.7.

![atom-gfm](https://cloud.githubusercontent.com/assets/1833684/9366047/95b43682-467a-11e5-828d-22e0874904a3.PNG)

A newline between `> text here` and *`'''`* highlights the remaining text correctly.

**Comment by fcharlie** [comment]
C++ syntax highlighting fails after 
![2015-10-15 14 39 24](https://cloud.githubusercontent.com/assets/6904176/10506500/2e5c534a-734b-11e5-8479-991b1e0dee75.png)

Atom 1.0.19 on Ubuntu 15.04

**Comment by 50Wliu** [comment]
@fcharlie that is a different bug.  Please see #44.

**Comment by cucumbur** [comment]
Similar issue here:
![screenshot 2017-02-03 15 37 48](https://cloud.githubusercontent.com/assets/8746469/22609451/dfa3ff42-ea26-11e6-9109-50a01c9ec14d.png)

**Comment by illpack** [comment]
Same problem here in Atom 1.14.4
Hints?
![capture](https://cloud.githubusercontent.com/assets/16364365/23749749/9b8b3e14-04c1-11e7-9d86-017e79376228.PNG)

**Comment by 50Wliu** [comment]
For now you have to completely close all your code snippets (in your case, make sure that they are valid Java).

**Comment by bric3** [comment]
@50Wliu In my opinion that can't be _valid_ if this is a snippet. The point of a snippet is to highlight part of a larger code base.

**Comment by bric3** [comment]
@50Wliu Another point, languages evolves. For example I have a valid Java code, but Atom does not recognise the valid syntax in the snippet, and as such it is not correctly closed.

In this code, the java language doesn't understand the try with resources (yet it's pretty old).

<img width="1065" alt="screen shot 2017-05-05 at 18 03 42" src="https://cloud.githubusercontent.com/assets/803621/25754325/069b4f44-31bf-11e7-8b54-4c37acfd9f1e.png">

**Comment by 50Wliu** [comment]
Yeah, I completely understand.  The problem is at this point we have _no way_ to force-close the code snippet.

(Also, try-with-resources support requires more work than it looks like)

**Comment by bric3** [comment]
@50Wliu Yes this problem seems to be related to this issue https://github.com/atom/language-java/issues/88.

In my opinion issues in the sub language plugin should not affect the gfm plugin. I am not a maintainer of this project so I dont know how hard it would be to do that.

**Comment by isiahmeadows** [comment]
BTW, I located the source of the bug in atom/first-mate#83 (missing feature, actually).

**Comment by illag** [comment]
welcome to bug world

**Comment by zalloy** [comment]
Looks like this bug hasn't been updated in a while. I'm running into this issue using Atom 1.40.1 on Linux. I've got this set of Markdown files that I'm using with Slate for some API documentation, and each of them has fenced code blocks for both Python and Javascript code examples. I noticed that in one of my files, the editor lost all the syntax highlighting after the first 2 code blocks, which is pretty annoying because it makes errors harder to spot. 

Any idea when this might get fixed?

**Comment by isiahmeadows** [comment]
Not sure, but I've been keeping close tabs on this personally as someone heavily affected by it. To my understanding, [Atom is migrating from a TextMate variant to Tree Sitter](https://flight-manual.atom.io/hacking-atom/sections/creating-a-grammar/), which solves it by design by basically [scoping](https://github.com/atom/first-mate/issues/83) all grammars by default. There doesn't appear to be a timeline for this AFAICT, but [IIUC, language-gfm can't be migrated to Tree Sitter until all its child grammars have Tree Sitter variants](https://github.com/atom/language-gfm/issues/214#issuecomment-482918153).

You could probably glean an idea how much progress has been made based on how many of the child Markdown grammars have been converted.

**Comment by boevski** [comment]
I've been hitting various problems of this type for years now.

How's that for an idea: Add an option in the plugin Settings to disable syntax highlighting for code snippets? This will take care of any bugs found in other Atom syntax libraries like Java. (Note that I still want the language identifier - ```java - as it is needed by my static site generator. I just don't want it to have any effect in Atom).

As an author, I don't rely too much on syntax-highlighted code snippets anyway as I can see the end result in the static site generator's output.
